extensions/cdc-mysql/src/main/java/com/hazelcast/jet/cdc/mysql/MySqlCdcSources.java

```diff
@@ -301,6 +352,42 @@ public Builder setSslTruststorePassword(@Nonnull String password) {
             return this;
         }
 
+        /**
+         * Interval in milliseconds after which to do periodic connection checking
+         * and initiate reconnect, if necessary. Defaults to
+         * {@value CdcSource#DEFAULT_RECONNECT_INTERVAL_MS} milliseconds.
+         */
+        @Nonnull
+        public Builder setReconnectIntervalMs(long intervalMs) {
+            config.setProperty(CdcSource.RECONNECT_INTERVAL_MS, intervalMs);
+            config.setProperty("connect.keep.alive.interval.ms", intervalMs);
+            return this;
+        }
+
+        /**
+         * Specifies how the connector should behave when it detects that the
+         * backing database has been shut down (note: temporary connection
+         * disruptions will not be interpreted in this way; after simple
+         * network outages the connector will automatically reconnect,
+         * regardless of this setting).
+         * <p>
+         * Possible values are (they are <em>not</em> case sensitive):
+         * <ul>
+         *     <li><em>FAIL</em>: will cause the whole job to fail</li>
+         *     <li><em>CLEAR_STATE_AND_RECONNECT</em>: will reconnect to
+         *      database, but will clear all internal state first, thus behaving
+         *      as if it would be connecting the first time (for example
+         *      snapshotting will be repeated)</li>
+         *     <li><em>RECONNECT</em>: will reconnect as is, in the same state
+         *      as it was at the moment of the disconnect </li>
+         * </ul>
+         */
+        @Nonnull
+        public Builder setReconnectBehaviour(String behaviour) {
```

Comment:
I don't want to expose that implementation code + this will probably be rewritten to a new abstraction taken from https://resilience4j.readme.io/docs/retry anyways.